cpp/Ice/greeter/ChatbotAMD.cpp

```diff
@@ -43,6 +45,15 @@ ServerAMD::Chatbot::greetAsync(
             response(os.str());
         }));
 
+    // We don't want the _tasks vector to grow forever so remove all completed task here, without waiting.
+    // TODO: switch to std::erase_if when we can use C++20.
```

Comment:
The pre-C++20 syntax is unfortunately not very readable. See item 32 in Scott Meyers' Effective STL.